### PR TITLE
#10117: Migrate fast_reduce_nc op to ttnn

### PR DESCRIPTION
**Pull request by cglagovichTT**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/10117

### Problem description
Move fast_reduce_nc to ttnn experimental

### Checklist
- [x] Post commit CI passes (https://github.com/tenstorrent/tt-metal/actions/runs/10358963925)
~- [ ] Blackhole Post commit (if applicable)~
- [ ] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: ttnn/cpp/ttnn/operations/experimental/reduction/fast_reduce_nc/fast_reduce_nc.hpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        const ttnn::MemoryConfig memory_config,
+        std::optional<const DeviceComputeKernelConfig> compute_kernel_config);
+};
+
+}  // namespace operations::experimental::reduction
+
+namespace experimental::reduction {
+
+constexpr auto fast_reduce_nc = ttnn::register_operation_with_auto_launch_op<
+    "ttnn::experimental::reduction::fast_reduce_nc",
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/experimental/reduction/fast_reduce_nc/fast_reduce_nc.hpp:37]:
```suggestion
    "ttnn::experimental::fast_reduce_nc",
```

Comment (ayerofieiev-tt) [reply ttnn/cpp/ttnn/operations/experimental/reduction/fast_reduce_nc/fast_reduce_nc.hpp:37]: It is not critical, but good for consistency. 
Similarly to how we register other ops in ttnn, we register ttnn::experimental::subcategory::op... as ttnn::exerimental::op

**Review by ayerofieiev-tt — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/experimental/reduction/fast_reduce_nc/fast_reduce_nc.hpp (hunk 1 above)
Comment (ayerofieiev-tt) [reply ttnn/cpp/ttnn/operations/experimental/reduction/fast_reduce_nc/fast_reduce_nc.hpp:37]: then need to adjust python calls too

Comment (cglagovichTT) [reply ttnn/cpp/ttnn/operations/experimental/reduction/fast_reduce_nc/fast_reduce_nc.hpp]: Thanks Artem. I changed the registration to experimental and updated python calls

**Review by mtairum — APPROVED** [review]

**Review by uaydonat — APPROVED** [review]

**Comment by cglagovichTT** [comment]
Rebase was a bit busy - pushed to separate branch and ran pipeline https://github.com/tenstorrent/tt-metal/actions/runs/10373553802

**Comment by cglagovichTT** [comment]
Pipeline https://github.com/tenstorrent/tt-metal/actions/runs/10373758076

**Comment by cglagovichTT** [comment]
Post commit passed except for a spurious resnet grayskull failure (CI file permission issue?) 
https://github.com/tenstorrent/tt-metal/actions/runs/10373758076

**Pull request merged**